packages/mui-base/src/InputUnstyled/useInput.test.tsx

```diff
@@ -0,0 +1,23 @@
+import * as React from 'react';
+import { createRenderer } from 'test/utils';
+import { expect } from 'chai';
+import { useInput } from '@mui/base/InputUnstyled';
+
+describe('useInput', () => {
+  const { render } = createRenderer();
+  describe('params: inputRef', () => {
+    it('should be able to attach input ref passed through params', () => {
+      const inputRef = React.createRef<HTMLInputElement>();
+
+      function Input() {
+        const { getInputProps } = useInput({
+          inputRef,
+        });
+        return <input {...getInputProps()} />;
+      }
+      const { getByRole } = render(<Input />);
+
+      expect(getByRole('textbox')).to.deep.equal(inputRef.current);
```

Comment:
This should be the other way around:
```suggestion
      expect(inputRef.current).to.deep.equal(getByRole('textbox'));
```

Reply:
Updated